odl/space/npy_tensors.py

```diff
@@ -0,0 +1,1992 @@
+        This function is part of the subclassing API. Do not
+        call it directly.
+
+        Parameters
+        ----------
+        x1, x2 : `NumpyTensor`
+            Factors in the product.
+        out : `NumpyTensor`
+            Element to which the result is written.
+
+        Examples
+        --------
+        >>> space = odl.rn(3)
+        >>> x = space.element([1, 0, 3])
+        >>> y = space.element([-1, 1, -1])
+        >>> space.multiply(x, y)
+        rn(3).element([-1.0, 0.0, -3.0])
+        >>> out = space.element()
+        >>> result = space.multiply(x, y, out=out)
+        >>> result
+        rn(3).element([-1.0, 0.0, -3.0])
+        >>> result is out
+        True
+        """
+        np.multiply(x1.data, x2.data, out=out.data)
+
+    def _divide(self, x1, x2, out):
+        """Compute the entry-wise quotient ``x1 / x2``.
+
+        This function is part of the subclassing API. Do not
+        call it directly.
+
+        Parameters
+        ----------
+        x1, x2 : `NumpyTensor`
+            Dividend and divisor in the quotient.
+        out : `NumpyTensor`
+            Element to which the result is written.
+
+        Examples
+        --------
+        >>> space = odl.rn(3)
+        >>> x = space.element([2, 0, 4])
+        >>> y = space.element([1, 1, 2])
+        >>> space.divide(x, y)
+        rn(3).element([2.0, 0.0, 2.0])
+        >>> out = space.element()
+        >>> result = space.divide(x, y, out=out)
+        >>> result
+        rn(3).element([2.0, 0.0, 2.0])
+        >>> result is out
+        True
+        """
+        np.divide(x1.data, x2.data, out=out.data)
+
+    def __eq__(self, other):
+        """Return ``self == other``.
+
+        Returns
+        -------
+        equals : bool
+            True if ``other`` is an instance of ``type(self)``
+            with the same `NumpyTensorSpace.shape`,
+            `NumpyTensorSpace.dtype`, `NumpyTensorSpace.order`
+            and `NumpyTensorSpace.weighting`, otherwise False.
+
+        Examples
+        --------
+        >>> space = odl.rn(3)
+        >>> same_space = odl.rn(3, exponent=2)
+        >>> same_space == space
+        True
+
+        Different `shape`, `exponent`, `dtype` or `order` all result in
+        different spaces:
+
+        >>> diff_space = odl.rn((3, 4))
+        >>> diff_space == space
+        False
+        >>> diff_space = odl.rn(3, exponent=1)
+        >>> diff_space == space
+        False
+        >>> diff_space = odl.rn(3, dtype='float32')
+        >>> diff_space == space
+        False
+        >>> space == object
+        False
+        >>> space_a = odl.rn((2, 3), order='A')  # default
+        >>> space_c = odl.rn((2, 3), order='C')  # default
+        >>> space_a == space_c
+        False
+
+        A `NumpyTensorSpace` with the same properties is considered
+        equal:
+
+        >>> same_space = odl.NumpyTensorSpace((2, 3), dtype='float64')
+        >>> same_space == space_a
+        True
+        """
```

Comment:
Well, for me the  guideline is 
```python
>>> np.array_equal([3], [[3]])
False
```
The two spaces clearly differ in the number of axes.